x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java

```diff
@@ -237,18 +237,7 @@ public long getEstimatedOperations() {
     }
 
     public boolean isAvailableWithLicense(XPackLicenseState licenseState) {
-        // Basic is always true
-        if (licenseLevel.equals(License.OperationMode.BASIC)) {
-            return true;
-        }
-
-        // The model license does not matter, Platinum license gets the same functions as the highest license
-        if (licenseState.isAllowedByLicense(License.OperationMode.PLATINUM)) {
-            return true;
-        }
-
-        // catch the rest, if the license is active and is at least the required model license
-        return licenseState.isAllowedByLicense(licenseLevel, true, false);
+        return licenseState.isAllowedByLicense(licenseLevel);
```

Comment:
`isAllowedByLicense(OperationMode)` calls `isAllowedByLicense(minimumMode, true, true)` which is different to the previous code.

Reply:
Thanks David. You are right that this call is different from the previous code. The difference is that the previous one does **not** allow trial license while the current one does. But their behaviours are actually not different when examined in the context of preceding code.

In the previous version, above this line, there is another check:
```java
if (licenseState.isAllowedByLicense(License.OperationMode.PLATINUM)) {
    return true;
}
```

If the license is trial mode, it will be allowed already by the above check and will never reach the second check, i.e. it makes no difference whether the 2nd `isAllowedByLicense` is called with `true` or `false` for trial license, since the 1st check short-curcuits the logic for trial license.

With above being said, there is indeed a behaviour difference: previously platinum license is considered as highest and is guaranteed to be allowed. With this change, only Enterprise is guaranteed to be allowed. While writing this, I think we can achieve the same behaviour by change the code to:
```java
return License.OperationMode.PLATINUM.compareTo(licenseLevel) <= 0 
    || licenseState.isAllowedByLicense(licenseLevel);
```

Reply:
Above suggest code snippet has a racing issue as the licenseState could change between two calls. To remove potential racing issue was a motivation for this PR. Sorry yet I suggested another one. A better one should be as the follows:
```java
final License.OperationMode minimumMode =
    License.OperationMode.PLATINUM.compareTo(licenseLevel) < 0 
        ? License.OperationMode.PLATINUM : licenseLevel;
return licenseState.isAllowedByLicense(minimumMode);
```

Reply:
Thanks for the explanation it makes sense, there is no need for the additional platinum level check what you have written is better for future proofing. Will you backport this to 7.7? 

@grabowskit do you know why we did not deem a trial licence sufficient for model inference?

Reply:
Just noticed the `v7.7.0` label is not applied. But yes I plan to backport this to 7.7.

Reply:
👍  

There is an argument to backport the trail licence change to 7.6.1 as it would be considered a bug if that is not what we intended. Thank for finding this

Reply:
If the intention is "trial license should always allow access", this bug does not really manifest itself in 7.6 since the the buggy code is never really triggered. If this is the case, I'd prefer to not backport this to 7.6.1 as it requires a number of other PRs.

But if the intention is to "block trial license for access", we do need fix it then. In this case, I'd prefer to have a separate PR to fix 7.6.1 specifically to avoid cascading backports. 

@tvernum Do you wanna comment on this? Thanks

Reply:
> If the intention is "trial license should always allow access"

A trial license _should_ allow all platinum features to be used (see for example https://www.elastic.co/trialextension and https://www.elastic.co/guide/en/elasticsearch/reference/current/start-trial.html#_description_15).  It's a mistake if this wasn't the case before.

Reply:
Thanks @droberts195 That's my understanding as well. In this case, the bug will never be triggered as it is short-circuited by its preceding check.